distro/exporter.go

```diff
@@ -59,19 +61,24 @@ func tracesExporter(l logr.Logger) traceExporterFunc {
 	return tef
 }
 
-func newOTLPTracesExporter(_ logr.Logger, c *exporterConfig) (trace.SpanExporter, error) {
+func newOTLPTracesExporter(l logr.Logger, c *exporterConfig) (trace.SpanExporter, error) {
 	ctx := context.Background()
 
-	splunkEndpoint := otlpRealmTracesEndpoint()
-	if splunkEndpoint != "" {
-		// Direct ingest to Splunk Observabilty Cloud using HTTP/protobuf.
-		return otlptracehttp.New(ctx,
-			otlptracehttp.WithEndpoint(splunkEndpoint),
-			otlptracehttp.WithURLPath(otlpRealmTracesEndpointPath),
-			otlptracehttp.WithHeaders(map[string]string{
-				"X-Sf-Token": c.AccessToken,
-			}),
-		)
```

Comment:
`SPLUNK_REALM` need to have precedence over `OTEL_EXPORTER_OTLP_PROTOCOL`. Otherwise if people set only `SPLUNK_REALM` then it would not configure the exporters to use the o11y endpoints. This would be a backwards-incomaptible change.